### ISSUE TITLE
[BUG] New volume stuck detaching

### ISSUE THREAD

**Issue by cjyar** [longhorn/longhorn#2012]
**Describe the bug**
Note this is arm64, using `master-arm64` image tags.

A newly created volume is detaching and unable to attach to a pod.

Pod events:
```
  Type     Reason              Age                   From                     Message
  ----     ------              ----                  ----                     -------
  Warning  FailedScheduling    9m36s                 default-scheduler        persistentvolumeclaim "test1" not found
  Warning  FailedScheduling    8m35s                 default-scheduler        persistentvolumeclaim "test1" not found
  Warning  FailedScheduling    8m28s                 default-scheduler        0/5 nodes are available: 5 pod has unbound immediate PersistentVolumeClaims.
  Warning  FailedScheduling    8m28s                 default-scheduler        0/5 nodes are available: 5 pod has unbound immediate PersistentVolumeClaims.
  Normal   Scheduled           8m7s                  default-scheduler        Successfully assigned default/test1-7f6c4d4545-8l4ll to node04
  Warning  FailedAttachVolume  114s (x11 over 8m7s)  attachdetach-controller  AttachVolume.Attach failed for volume "pvc-dc1e714a-be64-4ffc-994a-5a0ccbbad495" : rpc error: code = Aborted desc = The volume pvc-dc1e714a-be64-4ffc-994a-5a0ccbbad495 is detaching
  Warning  FailedMount         96s (x3 over 6m4s)    kubelet                  Unable to attach or mount volumes: unmounted volumes=[data], unattached volumes=[data default-token-wbd5m]: timed out waiting for the condition
```

PVC events:
```
  Type     Reason                 Age                From                                                                                      Message
  ----     ------                 ----               ----                                                                                      -------
  Normal   ExternalProvisioning   12m (x2 over 12m)  persistentvolume-controller                                                               waiting for a volume to be created, either by external provisioner "driver.longhorn.io" or manually created by system administrator
  Normal   Provisioning           12m (x2 over 12m)  driver.longhorn.io_csi-provisioner-686b6fff79-sj2dg_0d4e5093-7b79-4fdf-89e6-af210935fa2f  External provisioner is provisioning volume for claim "default/test1"
  Warning  ProvisioningFailed     12m                driver.longhorn.io_csi-provisioner-686b6fff79-sj2dg_0d4e5093-7b79-4fdf-89e6-af210935fa2f  failed to provision volume with StorageClass "longhorn": rpc error: code = DeadlineExceeded desc = context deadline exceeded
  Normal   ProvisioningSucceeded  12m                driver.longhorn.io_csi-provisioner-686b6fff79-sj2dg_0d4e5093-7b79-4fdf-89e6-af210935fa2f  Successfully provisioned volume pvc-dc1e714a-be64-4ffc-994a-5a0ccbbad495
```

**To Reproduce**
1. Download `deploy/longhorn.yaml` from master.
1. Replace every instance of `longhornio/${image}:{tag}` with `longhornio/${image}:master-arm64`.
1. Uncomment `CSI_ATTACHER_IMAGE` and set it to `longhornio/csi-attacher:v2.2.1-lh1`.
1. Uncomment `CSI_PROVISIONER_IMAGE` and set it to `longhornio/csi-provisioner:v1.6.0-lh1`.
1. Uncomment `CSI_NODE_DRIVER_REGISTRAR_IMAGE` and set it to `longhornio/csi-node-driver-registrar:v1.2.0-lh1`.
1. Uncomment `CSI_RESIZER_IMAGE` and set it to `longhornio/csi-resizer:v0.5.1-lh1`.
1. Create a PVC with `storageClassName: longhorn`.
1. Create a pod to mount the PVC.

**Expected behavior**
The pod should become Ready.

**Log**
If applicable, add the Longhorn managers' log when the issue happens.

**Environment:**
 - Longhorn version: f91466b
 - Kubernetes version: v1.19.3+k3s3
 - Node config
   - CPU type: arm64
   - OS: Ubuntu 20.04.1
   - CPU per node: 4
   - Memory per node: 4GB
   - Disk type: SD flash card
   - Network bandwidth and latency between the nodes: local 1Gbps switch
 - Underlying Infrastructure (e.g. on AWS/GCE, EKS/GKE, VMWare/KVM, Baremetal): Bare metal

**Additional context**
Fixed by modifying ClusterRole `longhorn-role` to add:
        ```
            - apiGroups: ["policy"]
              resources: ["poddisruptionbudgets"]
              verbs: ["get", "watch", "list", "delete", "update", "create"]
        ```

**Comment by yasker** [comment]
Can you share the log of `longhorn-manager`? As well as `longhorn-csi-plugin` log inside the `longhorn-csi-plugin` pod.

**Comment by cjyar** [comment]
The longhorn-manager log had lots of these, until I modified the ClusterRole:
```
time="2020-11-24T05:13:07Z" level=warning msg="Error syncing Longhorn instance manager longhorn-system/instance-manager-r-b133944b: fail to sync instance manager for longhorn-system/instance-manager-r-b133944b: poddisruptionbudgets.policy is forbidden: User \"system:serviceaccount:longhorn-system:longhorn-service-account\" cannot create resource \"poddisruptionbudgets\" in API group \"policy\" in the namespace \"longhorn-system\""
```

And the CSI pod had many repetitions of this:
```
time="2020-11-24T05:09:27Z" level=error msg="GRPC error: rpc error: code = Aborted desc = The volume pvc-dc1e714a-be64-4ffc-994a-5a0ccbbad495 is detaching"
time="2020-11-24T05:09:27Z" level=info msg="GRPC call: /csi.v1.Controller/ControllerPublishVolume"
time="2020-11-24T05:09:27Z" level=info msg="GRPC request: {\"node_id\":\"node04\",\"volume_capability\":{\"AccessType\":{\"Mount\":{\"fs_type\":\"ext4\"}},\"access_mode\":{\"mode\":1}},\"volume_context\":{\"fromBackup\":\"\",\"numberOfReplicas\":\"3\",\"staleReplicaTimeout\":\"2880\",\"storage.kubernetes.io/csiProvisionerIdentity\":\"1606192170543-8081-driver.longhorn.io\"},\"volume_id\":\"pvc-dc1e714a-be64-4ffc-994a-5a0ccbbad495\"}"
time="2020-11-24T05:09:27Z" level=info msg="ControllerServer ControllerPublishVolume req: volume_id:\"pvc-dc1e714a-be64-4ffc-994a-5a0ccbbad495\" node_id:\"node04\" volume_capability:<mount:<fs_type:\"ext4\" > access_mode:<mode:SINGLE_NODE_WRITER > > volume_context:<key:\"fromBackup\" value:\"\" > volume_context:<key:\"numberOfReplicas\" value:\"3\" > volume_context:<key:\"staleReplicaTimeout\" value:\"2880\" > volume_context:<key:\"storage.kubernetes.io/csiProvisionerIdentity\" value:\"1606192170543-8081-driver.longhorn.io\" > "
time="2020-11-24T05:09:27Z" level=error msg="GRPC error: rpc error: code = Aborted desc = The volume pvc-dc1e714a-be64-4ffc-994a-5a0ccbbad495 is detaching"
time="2020-11-24T05:14:27Z" level=info msg="GRPC call: /csi.v1.Controller/ControllerPublishVolume"
time="2020-11-24T05:14:27Z" level=info msg="GRPC request: {\"node_id\":\"node04\",\"volume_capability\":{\"AccessType\":{\"Mount\":{\"fs_type\":\"ext4\"}},\"access_mode\":{\"mode\":1}},\"volume_context\":{\"fromBackup\":\"\",\"numberOfReplicas\":\"3\",\"staleReplicaTimeout\":\"2880\",\"storage.kubernetes.io/csiProvisionerIdentity\":\"1606192170543-8081-driver.longhorn.io\"},\"volume_id\":\"pvc-dc1e714a-be64-4ffc-994a-5a0ccbbad495\"}"
time="2020-11-24T05:14:27Z" level=info msg="ControllerServer ControllerPublishVolume req: volume_id:\"pvc-dc1e714a-be64-4ffc-994a-5a0ccbbad495\" node_id:\"node04\" volume_capability:<mount:<fs_type:\"ext4\" > access_mode:<mode:SINGLE_NODE_WRITER > > volume_context:<key:\"fromBackup\" value:\"\" > volume_context:<key:\"numberOfReplicas\" value:\"3\" > volume_context:<key:\"staleReplicaTimeout\" value:\"2880\" > volume_context:<key:\"storage.kubernetes.io/csiProvisionerIdentity\" value:\"1606192170543-8081-driver.longhorn.io\" > "
time="2020-11-24T05:14:27Z" level=debug msg="ControllerPublishVolume: current nodeID node04"
```

**Comment by yasker** [comment]
I see. The deployment method is wrong.

The master for `github.com/longhonr/longhorn` is still v1.0.2. To deploy the in-dev master, you need to deploy from https://github.com/longhorn/longhorn-manager , See https://github.com/longhorn/longhorn-manager#deployment . You shouldn't need to modify anything to make it work.

You can use the uninstaller to cleanup the broken deployment for now.

**Comment by cjyar** [comment]
Thanks! It worked as you said.

**Comment by MalibuKoKo** [comment]
Hi, can we reopen this issue ?

I install longhorn with helm chart with following command :

```
helm upgrade -i longhorn longhorn/longhorn -n longhorn-system \
  --set image.longhorn.engineTag=master-arm64  \
  --set image.longhorn.managerTag=master-arm64  \
  --set image.longhorn.uiTag=master-arm64  \
  --set image.longhorn.instanceManagerTag=master-arm64  \
  --set csi.attacherImageTag=v2.2.1-lh1-arm64  \
  --set csi.provisionerImageTag=v1.6.0-lh1-arm64  \
  --set csi.nodeDriverRegistrarImageTag=v1.2.0-lh1-arm64  \
  --set csi.resizerImageTag=v0.5.1-lh1-arm64  \
  --set csi.snapshotterImageTag=v2.1.1-lh1-arm64
```

pod **longhorn-driver-deployer** stuck in **Init** status and pod **longhorn-manager** show same issue:
```
$ kubectl logs longhorn-manager-bndsx -c longhorn-manager --namespace=longhorn-system --tail=20 -f
I1202 22:47:19.930929       1 leaderelection.go:251] successfully acquired lease longhorn-system/longhorn-manager-upgrade-lock
time="2020-12-02T22:47:19Z" level=info msg="Start upgrading"
time="2020-12-02T22:47:19Z" level=info msg="No API version upgrade is needed"
time="2020-12-02T22:47:20Z" level=info msg="Finish upgrading"
time="2020-12-02T22:47:20Z" level=info msg="Upgrade leader lost: rakm01"
W1202 22:47:20.141227       1 client_config.go:541] Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.
I1202 22:47:20.149482       1 shared_informer.go:197] Waiting for caches to sync for longhorn datastore
E1202 22:47:20.198735       1 reflector.go:123] k8s.io/client-go/informers/factory.go:134: Failed to list *v1beta1.PodDisruptionBudget: poddisruptionbudgets.policy is forbidden: User "system:serviceaccount:longhorn-system:longhorn-service-account" cannot list resource "poddisruptionbudgets" in API group "policy" at the cluster scope
```

I fix it with the following command:
```
kubectl patch clusterroles.rbac.authorization.k8s.io longhorn-role --type='json' -p='[{"op": "add", "path": "/rules/1", "value": {"apiGroups": ["policy"], "resources": ["poddisruptionbudgets"], "verbs": ["get", "watch", "list", "delete", "update", "create"]} }]'
```

I think CRD is not fixed in the last version :
- REPO: https://charts.longhorn.io
- CHART: longhorn-1.0.2
- APP VERSION: v1.0.2

**Comment by yasker** [comment]
@MalibuKoKo ARM64 is not supported for Longhorn v1.0.2, and only replace the images won't work. Please install using `longhorn-manager` repo as I mentioned above if you want to give it a try before the formal release. ARM64 will be released with the upcoming Longhorn v1.1.0 release, with the Helm chart.
